processor/processor.go

```diff
@@ -3373,6 +3400,68 @@ func (proc *Handle) transformSrcDest(
 	}
 }
 
+func (proc *Handle) handleResponseForWarehouseTransformation(
+	ctx context.Context,
+	eventsToTransform []transformer.TransformerEvent,
+	pResponse transformer.Response,
+	commonMetaData *transformer.Metadata,
+	eventsByMessageID map[string]types.SingularEventWithReceivedAt,
+) {
+	if _, ok := whutils.WarehouseDestinationMap[commonMetaData.DestinationType]; !ok {
+		return
+	}
+	if len(eventsToTransform) == 0 || !proc.config.enableWarehouseTransformations.Load() {
+		return
+	}
+
+	transformStartAt := timeutil.Now()
+	wResponse := proc.warehouseTransformer.Transform(ctx, eventsToTransform, proc.config.transformBatchSize.Load())
+	proc.warehouseTransformerStats.responseTime.Since(transformStartAt)
+
+	logStartAt := timeutil.Now()
+	differingEvents := proc.warehouseTransDifferEvents(eventsToTransform, pResponse, wResponse, eventsByMessageID)
+	if err := proc.warehouseTransformer.Log(differingEvents, commonMetaData); err != nil {
+		proc.logger.Warnn("Failed to log events for warehouse transformation debugging", obskit.Error(err))
+	}
+	proc.warehouseTransformerStats.logTime.Since(logStartAt)
+}
+
+func (proc *Handle) warehouseTransDifferEvents(
+	eventsToTransform []transformer.TransformerEvent,
+	pResponse, wResponse transformer.Response,
+	eventsByMessageID map[string]types.SingularEventWithReceivedAt,
+) []types.SingularEventT {
+	// If the event counts differ, return all events in the transformation
+	if len(pResponse.Events) != len(wResponse.Events) || len(pResponse.FailedEvents) != len(wResponse.FailedEvents) {
+		events := lo.Map(eventsToTransform, func(e transformer.TransformerEvent, _ int) types.SingularEventT {
+			return eventsByMessageID[e.Metadata.MessageID].SingularEvent
+		})
+		proc.warehouseTransformerStats.mismatches.Count(len(events))
+		return events
+	}
+
+	var (
+		differedSampleEvents []types.SingularEventT
+		differedEventsCount  int
```

Comment:
Do we need a separate variable for events count ? It can be computed using `len(differedSampleEvents)`

Reply:
No, we are just taking one of the sample events in `differedSampleEvents`. `differedEventsCount` will contain all number of different events.